File: dotnet/azure/Mcma.Azure.BlobStorage/BlobStorageFolderLocator.cs
namespace Mcma.Azure.BlobStorage
{
    public class BlobStorageFolderLocator : BlobStorageLocator
    {
        public string FolderPath { get; set; }

        public override string Url => $"http://{StorageAccountName}.blob.core.windows.net/{Container}/{FolderPath}";
    }
}

File: azure/blob-storage/lib/blob-storage-folder-locator.ts
import { BlobStorageLocatorProperties, BlobStorageLocator } from "./blob-storage-locator";

export interface BlobStorageFolderLocatorProperties extends BlobStorageLocatorProperties {
    folderPath: string;
}

export class BlobStorageFolderLocator extends BlobStorageLocator implements BlobStorageFolderLocatorProperties {
    folderPath: string;
    
    constructor(properties: BlobStorageFolderLocatorProperties) {
        super("BlobStorageFolderLocator", properties);
        this.checkProperty("folderPath", "string");
    }

    get url(): string {
        return `https://${this.storageAccountName}.blob.core.windows.net/${this.container}/${this.folderPath}`;
    }
}
